File: api/wallee/transaction.go
package wallee

import (
	"fmt"
	"io"
	"net/http"
)

func (c *WalleeClient) CreateTransaction(tx *TransactionRequest) (*TransactionResponse, error) {
	path := fmt.Sprintf("/api/transaction/create?spaceId=%s", c.SpaceID)
	txr := &TransactionResponse{}
	err := c.jsonRequest(path, http.MethodPost, tx, &txr)
	return txr, err
}

func (c *WalleeClient) SearchTransaction(txreq *TransactionSearchRequest) ([]*TransactionResponse, error) {
	path := fmt.Sprintf("/api/transaction/search?spaceId=%s", c.SpaceID)
	txr := []*TransactionResponse{}
	err := c.jsonRequest(path, http.MethodPost, txreq, &txr)
	return txr, err
}

func (c *WalleeClient) CreatePaymentPageURL(transactionID int64) (string, error) {
	path := fmt.Sprintf("/api/transaction-payment-page/payment-page-url?spaceId=%s&id=%d", c.SpaceID, transactionID)

	req, err := c.request(path, http.MethodGet, nil)
	if err != nil {
		return "", err
	}

	resp, err := c.client.Do(req)
	if err != nil {
		return "", err
	}

	resp_bytes, err := io.ReadAll(resp.Body)
	if err != nil {
		return "", err
	}

	return string(resp_bytes), nil
}

File: public/js/api/transaction.js
import { protected_fetch } from "./protected_fetch.js";

export const get_all = () => protected_fetch(`api/transaction`);

export const get_by_id = id => protected_fetch(`api/transaction/${id}`);

export const create = data => protected_fetch(`api/transaction/create`, {
    method: "POST",
    body: JSON.stringify(data)
});

export const check = tx => protected_fetch(`api/transaction/${tx.id}/check`);

export const search_transaction = s => protected_fetch(`api/transaction/search`, {
    method: "POST",
    body: JSON.stringify(s)
});
